sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.internal.Utils;
+import io.opentelemetry.trace.Link;
+import io.opentelemetry.trace.Sampler;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Span.Builder;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.SpanContext;
+import java.util.ArrayList;
+import java.util.List;
+
+/** {@link SpanBuilderSdk} is SDK implementation of {@link Span.Builder}. */
+public class SpanBuilderSdk implements Span.Builder {
+
+  private SpanContext parent;
+  private Kind spanKind;
+  private List<Link> links;
+  private boolean recordEvents;
+  private Sampler sampler;
+  private final String spanName;
+  private boolean noParent;
+
+  SpanBuilderSdk(String spanName) {
+    this.spanName = spanName;
+  }
+
+  @Override
+  public Builder setParent(Span parent) {
+    Utils.checkNotNull(parent, "parent");
+    this.parent = parent.getContext();
```

Comment:
https://github.com/open-telemetry/opentelemetry-java/blob/bdfdeef7e720cdb5005ca09588724801912a4744/api/src/main/java/io/opentelemetry/trace/Span.java#L315-L317

Is this comment in the API doc valid? If the implementation only needs `SpanContext` anyway (and even calls `parent.getContext()`) we should probably remove it.

Reply:
Yes, the reason is that we want to propagate the https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk/src/main/java/io/opentelemetry/sdk/internal/TimestampConverter.java between parent and child. We create an TimestampConverter object in the first span in this process and propagate, then we record all the events using nanoTime and convert the nanoTime at export time in TimeStamp. This guarantees ordering of the events because nanoTime is monotonic.